### PR TITLE
fix: pin ahash to 0.8.0

### PR DESCRIPTION
**Pull request by mattbonnell**
0.8.1 has a bug https://github.com/tkaitchuck/aHash/issues/143

### CODE REVIEW

**Comment by tustvold** [comment]
I believe this pin be just added by impacted downstreams, i.e. those that are using custom build tooling. I'm not sure we need to pin this here?

**Comment by tustvold** [comment]
Yes

**Comment by mattbonnell** [comment]
> Yes

I had tried but it didn’t seem to be working. Will try again. 

Even so, ahash-8.0.1 uses cfg-target-abi, which is only available in nightly Rust. Doesn’t this then mean that installing arrow will fail for anyone not running a nightly, as it is for me?

**Comment by tustvold** [comment]
My understanding is it is just ignored outside of nightly

**Comment by mattbonnell** [comment]
> My understanding is it is just ignored outside of nightly

You mentioned this would affect folks using custom build tooling. I’m using Bazel, do you think that’s related?

**Comment by tustvold** [comment]
Yes, we had issues with hakari parsing the ahash crate, that @carols10cents tracked down to https://github.com/EmbarkStudios/cfg-expr/pull/52#issue-1436484665

My understanding is that some parsers don't correctly understand unstable cargo build attributes, and this causes custom build tooling to fail.

File: arrow/Cargo.toml [hunk 1]
```diff
@@ -38,10 +38,10 @@ path = "src/lib.rs"
 bench = false
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]
-ahash = { version = "0.8", default-features = false, features = ["compile-time-rng"] }
+ahash = { version = "0.8.0", default-features = false, features = ["compile-time-rng"] }
```
Comment (tustvold) [inline arrow/Cargo.toml:41]:
```suggestion
ahash = { version = "=0.8.0", default-features = false, features = ["compile-time-rng"] }
```

FYI you will need an exact version pin, cargo applies semver compatibility by default

**Comment by tustvold** [comment]
Closing this, as workarounds exist, and the change to ahash has been reverted - https://github.com/tkaitchuck/aHash/pull/145

**Pull request closed without merging**